File: backend/src/data-collection/collect-metrics.py
# backend/src/data-collection/collect-metrics.py

import requests
import psycopg2
from datetime import datetime

DATABASE_URL = 'postgresql://yourusername:yourpassword@localhost:5432/scroll_analytics'

def collect_metrics():
    response = requests.get('https://api.scroll.io/network/metrics')
    data = response.json()

    conn = psycopg2.connect(DATABASE_URL)
    cur = conn.cursor()
    query = """
        INSERT INTO metrics (timestamp, transaction_volume, block_time, gas_fee)
        VALUES (%s, %s, %s, %s)
    """
    values = (datetime.now(), data['txVolume'], data['blockTime'], data['gasFee'])
    cur.execute(query, values)
    conn.commit()
    cur.close()
    conn.close()

if __name__ == '__main__':
    collect_metrics()

File: backend/src/data-collection/collect-metrics.js
// backend/src/data-collection/collect-metrics.js

const axios = require('axios');
const { Client } = require('pg');

const client = new Client({
  user: 'yourusername',
  host: 'localhost',
  database: 'scroll_analytics',
  password: 'yourpassword',
  port: 5432,
});

async function collectMetrics() {
  try {
    await client.connect();
    const response = await axios.get('https://api.scroll.io/network/metrics');
    const data = response.data;

    const query = `
      INSERT INTO metrics (timestamp, transaction_volume, block_time, gas_fee)
      VALUES ($1, $2, $3, $4)
    `;
    const values = [new Date(), data.txVolume, data.blockTime, data.gasFee];

    await client.query(query, values);
    console.log('Metrics collected successfully');
  } catch (error) {
    console.error('Error collecting metrics:', error);
  } finally {
    await client.end();
  }
}

collectMetrics();
